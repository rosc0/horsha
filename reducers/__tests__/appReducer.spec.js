import reducer, { initialState as state } from '../app'
import types from '../../constants/actionTypes'

const expectResponses = {
  alert: {
    error: 'Test message',
    isDeviceConnected: null,
    isError: true,
  },
}

describe('App Reducer', () => {
  it('should return initial state', () => {
    expect(reducer()).toBe(state)
    expect(reducer()).toMatchSnapshot()
  })

  it('should SET_CONNECTION_STATUS', () => {
    expect(
      reducer(state, {
        type: types.SET_CONNECTION_STATUS,
        payload: true,
      })
    ).toMatchSnapshot()
  })

  it('should SET_ALERT_STATUS', () => {
    const act = reducer(state, {
      type: types.SET_ALERT_STATUS,
      config: 'Test message',
    })

    expect(act).toEqual(expectResponses.alert)
    expect(act).toMatchSnapshot()
  })

  it('should CLEAN_ALERT_STATUS', () => {
    expect(
      reducer(state, {
        type: types.CLEAN_ALERT_STATUS,
        payload: true,
      })
    ).toMatchSnapshot()
  })
})
