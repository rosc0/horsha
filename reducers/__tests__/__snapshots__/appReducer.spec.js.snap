// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Reducer should CLEAN_ALERT_STATUS 1`] = `
Object {
  "currentRouteName": undefined,
  "error": Object {
    "message": "",
    "title": "",
    "type": "",
  },
  "isDeviceConnected": null,
  "isError": false,
}
`;

exports[`App Reducer should SET_ALERT_STATUS 1`] = `
Object {
  "currentRouteName": undefined,
  "error": "Test message",
  "isDeviceConnected": null,
  "isError": true,
}
`;

exports[`App Reducer should SET_CONNECTION_STATUS 1`] = `
Object {
  "currentRouteName": undefined,
  "error": Object {
    "message": "",
    "title": "",
    "type": "",
  },
  "isDeviceConnected": undefined,
  "isError": false,
}
`;

exports[`App Reducer should return initial state 1`] = `
Object {
  "currentRouteName": undefined,
  "error": Object {
    "message": "",
    "title": "",
    "type": "",
  },
  "isDeviceConnected": null,
  "isError": false,
}
`;
